src/debug/debug.cc

```diff
@@ -3052,17 +3044,27 @@ static Handle<Object> RecordReplayConvertFunctionOffsetToLocation(Isolate* isola
     auto iter = gBreakpointPositions->find(key);
     if (iter == gBreakpointPositions->end()) {
       GenerateBreakpointInfo(isolate, script);
-
       iter = gBreakpointPositions->find(key);
-      if (iter == gBreakpointPositions->end()) {
-        recordreplay::Diagnostic("Unknown offset %s %d for RecordReplayConvertFunctionOffsetToLocation, crashing.",
-                                 function_id.c_str(), bytecode_offset);
-        IMMEDIATE_CRASH();
-      }
     }
 
-    line = iter->second.line_;
-    column = iter->second.column_;
+    if (iter != gBreakpointPositions->end()) {
+      line = iter->second.line_;
+      column = iter->second.column_;
+    } else {
+      recordreplay::Diagnostic("Unknown offset %s %d for RecordReplayConvertFunctionOffsetToLocation",
+                                function_id.c_str(), bytecode_offset);
+    }
+  }
+
+  // If there wasn't an offset or an unexpected unknown offset was encountered,
+  // fallback to the position of the function itself.
+  if (!line && !column) {
```

Comment:
Is line always guaranteed to be `>= 1` if it exists?  I'm assuming that's what you're relying on to distinguish the initialized state from uninitialized?  Might be good to add a note here indicating that if it's the case.

Reply:
I'll update the comment, but we only do the transformation in the fallback case below.  When we successfully look up a breakpoint position location the line has already been converted to be 1-indexed.